spec.md

```diff
@@ -1090,6 +1180,342 @@ message ControllerServiceCapability {
+  string id = 2;
+
+  // Identity information for the source volume. Note that create
+  // snapshot from snapshot is not supported here so the source has to
+  // be a volume. This field is REQUIRED.
+  string volume_id = 3;
+
+  // Timestamp when the point-in-time snapshot is taken on the storage
+  // system. This field is REQUIRED.
+  string timestamp = 4;
+
+  // The status of a snapshot. A snapshot could have the following
+  // statuses:
+  // UPLOADING - A snapshot is cut and is now being uploaded.
+  // READY - A snapshot is ready for use.
+  // Some cloud providers will upload the snapshot to a location in
+  // the cloud (i.e., an object store) after the snapshot is cut.
+  // Uploading is a long process that could take hours. At this phase,
+  // since the snapshot is already cut, a `thaw` operation can be
+  // performed so application can start running again (assuming a
+  // `freeze` operation was done before taking the snapshot). The
+  // status of the snapshot will become `READY` after the upload is
+  // complete.
+  // For cloud providers and storage systems that don't have the
+  // uploading process, the status should be `READY` after the snapshot
+  // is cut. `thaw` can be done when the status is `READY` in this case.
+  // A `CREATING` status is not included here because CreateSnapshot
+  // is synchronous and will block until the snapshot is cut.
+  // An `ERROR` status is also not included because there should be an
+  // gRPC error code if an error happens.
+  SnapshotStatus status = 5;
+}
+
+// The status of a snapshot. This parameter provides guidance to the CO
+// on what action can be taken in the process of snapshotting. Based on
+// this information, CO can issue repeated (idemponent) calls to
+// CreateSnapshot, monitor the response, and make decisions.
+// Note that CreateSnapshot is a synchronous call and it must block
+// until the snapshot is cut. If the cloud provider or storage system
+// does not need to upload the snapshot after it is cut, the status
+// returned by CreateSnapshot should be `READY`.
+// If the cloud provider or storage system needs to upload the snapshot
+// after the snapshot is cut, the status returned by CreateSnapshot
+// should be `UPLOADINDG`. In this case, another RPC GetSnapshot should
+// be called to wait for the upload to complete. GetSnapshot is also
+// synchronous and it will block until the upload process is complete.
+// The status returned by GetSnapshot should be `READY`.
+// An `ERROR` status is not included because there should be an gRPC
+// error code if an error happens.
+message SnapshotStatus {
+  enum Type {
+     UNKNOWN = 0;
+     // A snapshot is cut and is now being uploaded.
+     UPLOADING = 1;
+     // A snapshot is ready for use.
+     READY = 2;
+  }
+  Type type = 1;
+}
+```
+
+In the future, we may want to add `tags` (map<string, string> tags) in `CreateSnapshotRequest` to identify a snapshot.
```

Comment:
I'd remove this paragraph given it's not there.

Reply:
Ok.

Reply:
Removed.